### PR TITLE
Prototype implementation of #18

### PR DESCRIPTION
**Pull request by dyunov**
Current shortcomings:
- [ ] 1) Metaproperties not involved;
- [x] 2) Modules not involved;
- [x] 3) the percentage shows incorrectly (should be well above 90% for "commutative");
- [ ] 4) slight visual flaw in the property table.

### CODE REVIEW

**Comment by dyunov** [comment]
Another idea: make the property/ring lists also sortable by name or %.

**Comment by rschwiebert** [comment]
I'm very interested in seeing these changes but... I having trouble getting it to work locally!  I'll be looking at this in the days to come...

File: moduleapp/templates/moduleapp/property_detail.html [hunk 1]
```diff
@@ -68,7 +62,41 @@ <h3>Metaproperties:</h3>
...
+                </tr>
+                {% endfor %}
+            </table>
+        </div>
+    </div>
     {% endif %}
 </div>
 
 {% endblock %}
+
```
Comment (rschwiebert) [inline moduleapp/templates/moduleapp/property_detail.html:98]: please delete the extra lines

File: moduleapp/views.py [hunk 2]
```diff
@@ -74,22 +74,59 @@ def nullboolsort(x):
...
+        )
+        return self.render_to_response(context)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        has_count = ModuleProperty.objects.filter(property=context['object'], has=True).count()
-        lacks_count = ModuleProperty.objects.filter(property=context['object'], has=False).count()
-        context['has_count'] = has_count
-        context['lacks_count'] = lacks_count
+        #has_count = ModuleProperty.objects.filter(property=context['object'], has=True).count()
```
Comment (rschwiebert) [inline moduleapp/views.py:100]: Let's eliminate these two commented lines.

File: moduleapp/views.py [hunk 3]
```diff
@@ -74,22 +74,59 @@ def nullboolsort(x):
...
+                return x
+            else:
+                return -1
+
+        if sorttype == 's':
+            mod_join.sort(key=lambda x: nullboolsort(x[1]))
+        else:
+            mod_join.sort(key=lambda x: x[0].name.lower())
+
+        #context['has_count'] = has_count
```
Comment (rschwiebert) [inline moduleapp/views.py:124]: Eliminate these two commented-out lines

File: ringapp/SearchUtils.py [hunk 4]
```diff
@@ -157,3 +157,44 @@ def completeness_scores(include_commutative=False):
         if datum is None:
             data[key] = 0
     return data
+
+def completeness_scores_ring(commutative_only=False):
```
Comment (rschwiebert) [inline ringapp/SearchUtils.py:161]: Please rename to "property_completeness_scores"

File: ringapp/views.py [hunk 5]
```diff
@@ -256,15 +257,38 @@ class PropertyList(ListView):
...
+        return props
 
 
 class CommPropertyList(ListView):
     model = Property
     template_name = 'ringapp/commproperty_list.html'
 
     def get_queryset(self):
-        return Property.objects.filter(commutative_only=True).order_by('name')
+        props = Property.objects.filter(commutative_only=True).order_by('name')
```
Comment (rschwiebert) [inline ringapp/views.py:282]: At the top of the file,
from django.db.models.functions import Lower
Then on L282, order_by(Lower('name')). 

This gets rid of the case-based sorting issue.

Comment (dyunov) [reply ringapp/views.py:282]: @rschwiebert Should I do the same for all 6 lists?

Comment (dyunov) [reply ringapp/views.py:282]: And there are also other `order_by`'s in the file, as in L374.

**Comment by rschwiebert** [comment]
I think several lists already have this behavior. To check just visit the
page in the browser and it should be obvious which ones need attention.

**Pull request merged**